app_flutter/lib/service/appengine_cocoon.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:fixnum/fixnum.dart';
+
+import 'package:cocoon_service/protos.dart'
+    show Commit, CommitStatus, Stage, Task;
+
+import 'cocoon.dart';
+
+/// CocoonService for interacting with flutter/flutter production build data.
+///
+/// This queries API endpoints that are hosted on AppEngine.
+class AppEngineCocoonService implements CocoonService {
+  /// The Cocoon API endpoint to query
+  ///
+  /// This is the base for all API requests to cocoon
+  static const baseApiUrl = 'https://flutter-dashboard.appspot.com/api';
```

Comment:
These fields could probably be private

Reply:
baseApiUrl can, but http.Client needs to be accessible for tests to inject a mock

Reply:
Rather than having a setter, I would make the constructor take an optional `http.Client` argument, defaulting to the regular client if one isn't provided. Since you're testing the implementation, you should still be able to inject the right value:

```
class AppEngineCocoonService {

  /// Creates a new [AppEngineCocoonService].
  ///
  /// If a [client] is not specified, a new [http.Client] instances is created.
  AppEngineCocoonService({http.Client client})
    _client = client ?? http.Client();
}

```

Reply:
This reduces the API surface, but still lets you test with a fake implementation.

Reply:
Thank you! I like the simple solution :)